vault/resource_gcp_auth_backend.go

```diff
@@ -165,37 +190,46 @@ func gcpAuthBackendConfigPath(path string) string {
 	return "auth/" + strings.Trim(path, "/") + "/config"
 }
 
-func gcpAuthBackendWrite(d *schema.ResourceData, meta interface{}) error {
+func gcpAuthBackendWrite(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	client, e := provider.GetClient(d, meta)
 	if e != nil {
-		return e
+		return diag.FromErr(e)
 	}
 
 	authType := gcpAuthType
-	path := d.Get("path").(string)
-	desc := d.Get("description").(string)
-	local := d.Get("local").(bool)
+	path := d.Get(consts.FieldPath).(string)
+	desc := d.Get(consts.FieldDescription).(string)
+	local := d.Get(consts.FieldLocal).(bool)
+
+	config := &api.MountConfigInput{}
+	useAPIver117Ent := provider.IsAPISupported(meta, provider.VaultVersion117) && provider.IsEnterpriseSupported(meta)
+	if useAPIver117Ent {
+		if v, ok := d.GetOk(consts.FieldIdentityTokenKey); ok {
+			config.IdentityTokenKey = v.(string)
+		}
+	}
 
 	log.Printf("[DEBUG] Enabling gcp auth backend %q", path)
-	err := client.Sys().EnableAuthWithOptions(path, &api.EnableAuthOptions{
+	err := client.Sys().EnableAuthWithOptionsWithContext(ctx, path, &api.EnableAuthOptions{
 		Type:        authType,
 		Description: desc,
 		Local:       local,
+		Config:      *config,
 	})
 	if err != nil {
-		return fmt.Errorf("error enabling gcp auth backend %q: %s", path, err)
+		return diag.Errorf("error enabling gcp auth backend %q: %s", path, err)
 	}
 	log.Printf("[DEBUG] Enabled gcp auth backend %q", path)
 
 	d.SetId(path)
 
-	return gcpAuthBackendUpdate(d, meta)
+	return gcpAuthBackendUpdate(ctx, d, meta)
 }
 
-func gcpAuthBackendUpdate(d *schema.ResourceData, meta interface{}) error {
+func gcpAuthBackendUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
```

Comment:
maybe out of scope for this PR, but should we pass the context to the Tune call on line 271? I am not sure if we have a helper for that already. We could leave a TODO comment on L271 if we don't want to address it in this PR.

Reply:
Good call, added a TODO note